hq_superset/models.py

```diff
@@ -0,0 +1,87 @@
+import uuid
+from dataclasses import dataclass
+from typing import Any, Literal
+
+from datetime import datetime
+from superset import db
+from authlib.integrations.sqla_oauth2 import OAuth2ClientMixin
+from werkzeug.security import check_password_hash, generate_password_hash
+
+
+@dataclass
+class DataSetChange:
+    action: Literal['upsert', 'delete']
+    data_source_id: str
+    data: dict[str, Any]
```

Comment:
I wonder if we might not want to have another dataclass for this `data` field. That way it will do its own validation and we will know exactly what keys we expect